ref/Microsoft.Build.Utilities.Core/net/Microsoft.Build.Utilities.Core.cs

```diff
@@ -434,6 +434,7 @@ public partial class TaskLoggingHelper : System.MarshalByRefObject
         public void LogWarningWithCodeFromResources(string messageResourceName, params object[] messageArgs) { }
         public void LogWarningWithCodeFromResources(string subcategoryResourceName, string file, int lineNumber, int columnNumber, int endLineNumber, int endColumnNumber, string messageResourceName, params object[] messageArgs) { }
         public void MarkAsInactive() { }
+        public bool ShouldLogMessage(Microsoft.Build.Framework.MessageImportance importance) { throw null; }
```

Comment:
I've been thinking, should we just expose the verbosity from the engine, and have an extension method ShouldLogMessage that reads that verbosity and compares with the one the task passes in?

Reply:
We would still have to look at verbosity in the logger, since you could attach multiple loggers with different verbosities.

Finding the maximum verbosity of any logger and revealing it through this ShouldLogMessage method seems about the best we can do.

Reply:
I have exposed it on `IBuildEngine` (specifically through `IBuildEngine10` -> `BuildEngineInterface` interface) and left it on `TaskLoggingHelper` as well for convenience.